### ISSUE TITLE
Roslyn should lear nullability from switches over tuples just as it does from tuple variables

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#37136]
**Steps to Reproduce**:

Compile the following code:
```cs
#nullable enable
class C
{
    void M(string? s1, string? s2)
    {
        var t = (s1, s2);
        // this works: t.Item1 becomes non-nullable within the switch section
        switch (t)
        {
            case ("", ""):
                t.Item1.ToString(); // no warnings
                break;
        }

        // this doesn't
        switch (s1, s2) // switch ((s1, s2)) doesn't work either
        {
            case ("", ""):
                s1.ToString(); // warning CS8602: Dereference of a possibly null reference.
                break;
        }
    }
}
```

**Expected Behavior**:
No warnings. The switch checks that both variables are not null in its single section.

**Actual Behavior**:
`warning CS8602: Dereference of a possibly null reference.` is reported for `s1.ToString()`

**Version Used**: 

```
Branch master (21 Jun 2019)
Latest commit 898bed by Heejae Chang:
added NFW to get some data on incremental parsing bug where source si� (#36620)

* added NFW to get some data on incremental parsing bug where source size and tree size is different

* more comments
```

**Comment by jcouv** [comment]
Verified this is fixed by https://github.com/dotnet/roslyn/pull/51001 (merged for 16.10)
I'll close the issue. Thanks
